### PR TITLE
[G2M] a few lint rule updates

### PR DESCRIPTION
**Pull request by IYCI**
adding `node` env to eslint config because https://github.com/storybooks/storybook/issues/3656

### CODE REVIEW

**Comment by woozyking** [comment]
Can we also add the rule to enforce no trailing white spaces?

File: stories/data.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ const largeDataset = [
   {
     'id':13,
     'first_name':'Sher',
-    'last_name':"O'Fogarty",
+    'last_name':'O\'Fogarty',
```
Comment (woozyking) [inline stories/data.js]: I think `quotes: ["single", { "avoidEscape": true }]`  is good for a "more natural" compromise to keep mostly single quotes, but not resort to readability hurdles like escape backslash.

Comment (IYCI) [reply stories/data.js]: we can apply that to snoke as well

Comment (woozyking) [reply stories/data.js]: At some point we should have a common repo for eslint rc so we can have a common baseline.

File: .eslintrc.js [hunk 2]
```diff
@@ -55,6 +55,12 @@ module.exports = {
     ],
     'key-spacing': [
       'error',
+    ],
+    'keyword-spacing': [
+      'error',
+    ],
+    'react/jsx-tag-spacing': [
```
Comment (woozyking) [inline .eslintrc.js]: oh this one addresses `<Component/>`?

Comment (IYCI) [reply .eslintrc.js]: yeah

**Pull request merged**